qt/scientific_interfaces/ISISReflectometry/Reduction/BackgroundSubtraction.h

```diff
@@ -0,0 +1,99 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_CUSTOMINTERFACES_BACKGROUNDSUBTRACTION_H_
+#define MANTID_CUSTOMINTERFACES_BACKGROUNDSUBTRACTION_H_
+#include "Common/DllConfig.h"
+#include <boost/optional.hpp>
+#include <stdexcept>
+#include <string>
+namespace MantidQt {
+namespace CustomInterfaces {
+namespace ISISReflectometry {
+enum class BackgroundSubtractionType {
+  PerDetectorAverage,
+  Polynomial,
+  AveragePixelFit
+};
+enum class CostFunctionType { LeastSquares, UnweightedLeastSquares };
+
+inline BackgroundSubtractionType
+backgroundSubtractionTypeFromString(std::string const &subtractionType) {
+  if (subtractionType.empty() || subtractionType == "PerDetectorAverage")
```

Comment:
This might be a stylistic thing, but I was wondering if it might be worth creating a boost:bimap between the enum values and the strings?  Then most of these functions could be replaced with a simple value lookup.

Reply:
Good call - I hadn't come across bimap but that looks like a much better solution. I've added this to #26539 to do as a separate PR because there are several similar functions already and I'd like to do them all in one go.